AstroObject.cpp

```diff
@@ -61,42 +71,50 @@ int AstroObject::getMass()
 	return this->mass;
 }
 
+void AstroObject::setMass(int mass)
+{
+	this->mass = mass;
+}
+
 
-Rnfpp AstroObject::getPosition()
+Vector3 AstroObject::getPosition()
 {
 	return this->position;
 }
 
 
-void AstroObject::setPosition(Rnfpp position)
+void AstroObject::setPosition(Vector3 position)
 {
 	this->position = position;
 }
 
 
 void AstroObject::iter()
 {
+	
 	for(AstroObject & elem : this->affectedObjects)
 	{
 		if(this->position == elem.getPosition()) continue;
 		int distance = this->calculateDistance(elem.getPosition());
 		if (distance > this->influenceRadius)
 			continue;
 
-		Rnfpp direction = elem.getPosition() - this->position;
-		direction.normalise();
-		Rnfpp force = this->calculateForce(distance, direction);
+		Vector3 direction = elem.getPosition() - this->position;
+		direction.normalized();
+		Vector3 force = this->calculateForce(distance, direction);
 		elem.applyForce(force);
 	}
+	
 }
 
 
-int AstroObject::calculateDistance(Rnfpp position)
+int AstroObject::calculateDistance(Vector3 position)
 {
-	float distance = this->position.distance(position);
+	
+	float distance = this->position.distance_to(position);
 	//calculate actual distance
 	//...
-	return distance;
+	return (int)distance;
```

Comment:
If this function is intended to return an `int` in any case, why casting here?  Furthermore, there are also other possibilities to cast variables, for instance the functions therefore from the standard library; why they do not work here?